Drop DeleteSnapshot Event Handler

lkiesow opened the pull request:
This patch removes the event triggered after deleting a snapshot. This can easily be triggered thousands of times when cleaning up and nothing in Opencast is actually using the event.

This fixes #5173

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

lkiesow committed:
Drop DeleteSnapshot Event Handler

This patch removes the event triggered after deleting a snapshot. This
can easily be triggered thousands of times when cleaning up and nothing
in Opencast is actually using the event.

This fixes #5173

lkiesow committed:
Rename DeleteSnapshotHandler to DeleteEpisodeHandler

After all, it now only handles deletion of episodes.
The name was already confusing before.

Arnei commented on modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/query/AbstractADeleteQuery.java:
```diff
@@ -90,7 +90,7 @@ public AbstractADeleteQuery(AssetManagerImpl am, String owner) {
   }
 
-  public long run(DeleteSnapshotHandler deleteSnapshotHandler) {
+  public long run(DeleteEpisodeHandler deleteSnapshotHandler) {
```
```suggestion
  public long run(DeleteEpisodeHandler deleteEpisodeHandler) {
```

The one thing that is still called `deleteSnapshotHandler` or similar.

lkiesow replied:
Fixed

lkiesow committed:
Rename handler to what they actually are

This patch fixes naming of some variables which are based on renamed
class names.

KatrinIhler commented:
Discussed and approved, and a good idea in general. LGTM.
